cmake/compiler/gcc/target_security.cmake

```diff
@@ -0,0 +1,20 @@
+# See root CMakeLists.txt for description and what is expected of toolchain_cc_security
+macro(toolchain_cc_security)
+
+  if(NOT CONFIG_NO_OPTIMIZATIONS)
+    zephyr_compile_definitions(
+      _FORTIFY_SOURCE=2
```

Comment:
_FORTIFY_SOURCE is a feature of glibc[0], and only a feature of glibc[citation needed].

glibc is a libc implementation that is not used much in Zephyr due to it not being appropriate for resource-constrained embedded, the only platform that I know can use it is 'native_posix'.

Clang does not use glibc[1], so the clang toolchain should not be sourcing a file that enables a glibc feature.

The cleanest way to solve this would be to have the build system somehow detect that glibc is in use and it's version. And then have it enable _FORTIFY_SOURCE on the condition that we have a glibc implemntation that supports _FORTIFY_SOURCE. But I guess this is out-of-scope for this PR.

But having toolchain files only include the build scripts that apply to them is definitely within scope.

[0] https://gcc.gnu.org/ml/gcc-patches/2004-09/msg02055.html
[1] https://github.com/mongodb/mongo/blob/de4f60966b2ea5f461ba532edda65c560002140d/SConstruct#L2802

Reply:
Agree.